### PR TITLE
Added hook to allow extending of gutenberg editor in plugin/themes.

### PR DESCRIPTION
**Pull request by selul**
Ref: #1184

### CODE REVIEW

File: lib/client-assets.php [hunk 1]
```diff
@@ -371,5 +371,12 @@ function gutenberg_scripts_and_styles( $hook ) {
 		array( 'wp-components', 'wp-blocks' ),
 		filemtime( gutenberg_dir_path() . 'editor/build/style.css' )
 	);
+
+	global $hook_suffix;
+	/**
+	 * Hook to allow extending of gutenberg editor in plugin/themes.
+	 */
+	do_action( 'block_enqueue_script', $hook_suffix );
```
Comment (westonruter) [inline lib/client-assets.php]: This should be plural.

Comment (westonruter) [inline lib/client-assets.php]: Is `$hook_suffix` needed?

Comment (westonruter) [inline lib/client-assets.php]: Also, I think this should actually be `block_edit_enqueue_scripts` to make it clear that the scripts and styles being enqueued are for the _editor_ and not for blocks that are being rendered on the frontend.

Comment (youknowriad) [inline lib/client-assets.php]: I personal prefer `block_enqueue_scripts`. I'm guessing the frontend scripts and styles won't be enqueued using actions like this, but more with something tied to the block API.

Comment (selul) [inline lib/client-assets.php]: @westonruter i tried to replicate the `admin_enqueue_script` behaviour, which passes the $hook_sufix when calling the action.

**Comment by gziolo** [comment]
It was implemented and iterated a few times. You can see the latest docs here: https://github.com/WordPress/gutenberg/blob/master/docs/blocks-basic.md#enqueuing-block-scripts. I'm closing this PR, feel free to open another PR if you feel like we need more flexibility there.

**Pull request closed without merging**